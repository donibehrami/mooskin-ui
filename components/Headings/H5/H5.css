@import url('https://cdn.moostaging.com/assets/fonts/google-web-fonts/fonts.css');
@import '../../_utils/globals/variables.css';

.h5 {
    font-family: 'Convergence', 'Alegreya Sans', sans-serif;
    /*font-size: 12px;*/
    font-size: 0.75rem;
    color: var(--smallHeading);
    margin: 19px 0;
}
