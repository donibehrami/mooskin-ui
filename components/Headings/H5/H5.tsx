import * as React from 'react';

import styles from './H5.css';

import {IHeadingProps} from '../props';

class H5 extends React.Component<IHeadingProps, {}> {

    static defaultProps = {
        className: '',
        style: {}
    };

    static displayName = 'H5';

    render() {

        const {id, className, style, children} = this.props;

        const classes = `h5-component ${styles.h5} ${className}`;

        return (
            <div
                id={id}
                className={classes}
                style={style}
            >
                {children}
            </div>
        );
    }
}

export default H5;
