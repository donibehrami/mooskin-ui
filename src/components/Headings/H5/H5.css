@import url('https://fonts.googleapis.com/css?family=Convergence');

.h5 {
    font-family: 'Convergence', sans-serif;
    font-weight: 400;
    color: #3a4257;
}