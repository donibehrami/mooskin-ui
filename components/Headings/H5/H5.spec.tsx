import * as React from 'react';
import H5 from './H5';

import { shallow } from 'enzyme';

describe('H5', () => {

    test('renders correctly', () => {

        const tree = shallow(
            <H5
                className="myClass"
                style={{color: 'blue'}}
                id={'heading5'}
            >
                Mooskin
            </H5>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders properly into dom with text, id and class', () => {

        const component = shallow(<H5 id="Campaigns" className="headers">Mooo!</H5>);

        expect(component.find('p').text()).toBe('Mooo!');
        expect(component.find('p').prop('id')).toEqual('Campaigns');
        expect(component.find('p').hasClass('headers')).toBe(true);
    });

});
