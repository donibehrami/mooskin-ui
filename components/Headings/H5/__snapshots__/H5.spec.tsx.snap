// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H5 renders correctly 1`] = `
<p
  className="h5-component h5 myClass"
  id="heading5"
  style={
    Object {
      "color": "blue",
    }
  }
>
  Mooskin
</p>
`;
